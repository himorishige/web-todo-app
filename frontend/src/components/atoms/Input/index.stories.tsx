/** @jsxImportSource @emotion/react */
import { ComponentStory, ComponentMeta } from '@storybook/react';
import Input from '.';

export default {
  title: 'WebToDoApp/Atoms/Input',
  component: Input,
} as ComponentMeta<typeof Input>;

const Template: ComponentStory<typeof Input> = (args) => <Input {...args} />;

export const Default = Template.bind({});
Default.storyName = 'Default';
Default.args = {
  placeholder: 'タスクを入力してください',
  disabled: false,
  label: 'taskName',
  //@ts-ignore
  register: () => {},
};
