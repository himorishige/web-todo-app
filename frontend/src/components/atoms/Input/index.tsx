/** @jsxImportSource @emotion/react */
import { css } from '@emotion/react';
import { memo } from 'react';
import { Path, UseFormRegister } from 'react-hook-form';

type Inputs = {
  taskName: string;
  taskMemo: string;
};

export type Props = {
  placeholder: string;
  defaultValue?: string;
  disabled?: boolean;
  maxLength?: number;
  label: Path<Inputs>;
  register: UseFormRegister<Inputs>;
  required: boolean;
};

const Input: React.VFC<Props> = memo(
  ({
    placeholder,
    defaultValue,
    disabled = false,
    maxLength = 40,
    label,
    register,
    required,
    ...props
  }: Props) => {
    return (
      <input
        type="text"
        css={inputStyle}
        placeholder={placeholder}
        defaultValue={defaultValue}
        disabled={disabled}
        maxLength={maxLength}
        {...register(label, { required })}
      />
    );
  },
);

export default Input;

const inputStyle = css`
  -webkit-appearance: none;
  width: 100%;
  padding: 1rem;
  border: 2px solid var(--primary-color);
  border-radius: 4px;
  &:disabled {
    background: #f8f8f8;
  }
`;
