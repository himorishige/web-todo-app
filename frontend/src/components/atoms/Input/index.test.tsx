import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Input from '.';

describe('Atoms/Input', () => {
  const placeholder = '入力';
  const label = 'taskMemo';
  const register = jest.fn();
  const required = false;
  test('placeholderにPropsが表示される', () => {
    render(
      <Input placeholder={placeholder} label={label} register={register} required={required} />,
    );
    const inputElement = screen.getByRole('textbox') as HTMLInputElement;
    expect(inputElement.placeholder).toBe(placeholder);
  });
  test('テキストフィールドに1を入力したら1が表示される', () => {
    render(
      <Input placeholder={placeholder} label={label} register={register} required={required} />,
    );
    const inputElement = screen.getByRole('textbox') as HTMLInputElement;
    fireEvent.change(inputElement, { target: { value: '1' } });
    expect(inputElement.value).toBe('1');
  });
  test('正しくレンダリングされている', () => {
    const inputField = renderer
      .create(
        <Input placeholder={placeholder} label={label} register={register} required={required} />,
      )
      .toJSON();
    expect(inputField).toMatchSnapshot();
  });
});
