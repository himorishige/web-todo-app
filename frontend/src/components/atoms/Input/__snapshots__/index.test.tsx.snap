// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Input 正しくレンダリングされている 1`] = `
<input
  className="css-1gxkf0o-Input"
  data-testid="input-area"
  disabled={false}
  maxLength={40}
  placeholder="入力"
  type="text"
/>
`;
